llvm/include/llvm/Analysis/TargetLibraryInfo.def

```diff
@@ -462,11 +462,91 @@ TLI_DEFINE_ENUM_INTERNAL(atomic_load)
 TLI_DEFINE_STRING_INTERNAL("__atomic_load")
 TLI_DEFINE_SIG_INTERNAL(Void, SizeT, Ptr, Ptr, Int)
 
+/// int8_t __atomic_load_1(void *ptr, int memorder);
```

Comment:
Is the choice to let `__atomic_load_N` return a sized integer, while `__atomic_load` return void based on maintaining proximity with GCC's definitions of these builtins? Or is there any other reason which I am missing?

Reply:
This is the GCC-defined ABI, see https://gcc.gnu.org/wiki/Atomic/GCCMM/LIbrary#GCC_intrinsics for reference. 

`__atomic_load_N` returns the load value, possible due to having a known size that fits into a register (or pre-allocated stack slot, depending on calling convention). `__atomic_load` reads a dynamic number of bytes and hence cannot be pre-allocated/passed by register.